app/models/early_allocation.rb

```diff
@@ -1,13 +1,17 @@
 # frozen_string_literal: true
 
 class EarlyAllocation < ApplicationRecord
+  before_validation :record_outcome
```

Comment:
think this should be before_save ?

Reply:
@starswan If I change it to before_save I can't have the validation as it will fail automatically due to the validation. Other option is to remove the validation, but then you will have an outcome set to nil. Unless you have another idea?

Reply:
outcome should never be nil on save - it should be 1 of the 3 valid values. Its a little confusing, but the EA can only be in 1 or 3 (5?) states - auto_eligible, discretionary or ineligible. (extra 2 for discretionary true/false).